src/components/detail/GraphPanel.vue

```diff
@@ -0,0 +1,172 @@
+            </h5>
+        </div>
+        <div class="ui center aligned subdued basic segment">
+            <p>* Definition of {{metric}} goes here, pulled from config or maybe dynamically from the wiki page.</p>
+            <p>More descriptor text will go here assuming that it takes a few sentences to explain a term for a metric.</p>
+
+        </div>
+        <div class="ui right floated icon button" @click="toggleFullscreen">
+            <i class="ui icon" :class="{expand: !fullscreen, compress: fullscreen}"/>
+        </div>
+    </div>
+</section>
+</template>
+
+<script>
+import ArrowIcon from '../ArrowIcon'
+import TimeRangeSelector from '../TimeRangeSelector'
+import SimpleLegend from './SimpleLegend'
+import BarChart from './chart/BarChart'
+import LineChart from './chart/LineChart'
+import MapChart from './chart/MapChart'
+import TableChart from './chart/TableChart'
+import EmptyChart from './chart/EmptyChart'
+
+export default {
+    name: 'graph-panel',
+    components: {
+        ArrowIcon,
+        TimeRangeSelector,
+        SimpleLegend,
+        BarChart,
+        LineChart,
+        MapChart,
+        TableChart,
+        EmptyChart
+    },
+    props: ['metricData', 'wiki', 'breakdowns', 'fullscreen', 'graphModel'],
+    computed: {
+        breakdown: function () {
+            return (this.breakdowns || []).find((m) => m.on)
+        },
+        chartTypes: function () {
+            return this.getChartTypes();
+        },
+        chartIcon: function () {
+            return this.getChartTypes()[0].icon;
+        },
+        chartComponent: function () {
+            if (this.chartType) return this.chartType + '-chart'
+            let chartTypes = this.getChartTypes();
+            return (chartTypes[0].chart || 'empty') + '-chart'
+        },
+        total: function () {
+            const data = this.graphModel && this.graphModel.getGraphData();
+            if (!data) { return 0 }
+
+            return data.reduce((r, m) => r + m.total, 0)
+        },
+        metric: function () {
+            return this.$route.params.metric ?
+                this.$route.params.metric : this.defaultMetrics[this.area]
+        }
+    },
+    data () {
+        return {
+            chartType: null,
+            availableChartTypes: [
+                { chart: 'bar', icon: 'bar' },
+                { chart: 'line', icon: 'line' },
+                { chart: 'map', icon: 'globe' },
+                { chart: 'table', icon: 'table' },
+            ]
+        }
+    },
+    methods: {
+        changeChart (t) {
+            this.chartType = t.chart
+        },
+        changeTimeRange (range) {
+            this.$emit('changeTimeRange', range);
+        },
+        getChartTypes () {
+            return this.availableChartTypes.filter((c) => {
+                if (!this.metricData) { return false; }
```

Comment:
this whole mess should be configured somewhere, like a metrics to charts config or something.